src/main/java/org/kuali/kra/protocol/personnel/PersonnelHelperBase.java

```diff
@@ -100,8 +101,12 @@ public ProtocolPersonBase getNewProtocolPerson() {
     }
 
     public List<ProtocolUnitBase> getNewProtocolPersonUnits() {
-        if (getForm().getProtocolDocument().getProtocol().getProtocolPersons().size() > this.newProtocolPersonUnits.size()) {
-            this.newProtocolPersonUnits.add(this.newProtocolPersonUnits.size(), createNewProtocolUnitInstanceHook());
+        //UAR-2898: Initialize arrays to avoid errors on PersonUnitsSection edit
+        if (newProtocolPersonUnits == null || newProtocolPersonUnits.isEmpty() ||
+                getForm().getProtocolDocument().getProtocol().getProtocolPersons().size() > this.newProtocolPersonUnits.size()) {
+            for (int idx=newProtocolPersonUnits.size(); idx<getForm().getProtocolDocument().getProtocol().getProtocolPersons().size(); idx++) {

```

Comment:
If newProtocolPersonUnits is null, the int idx=newProtocolPersonUnits.size() throw a NPE?

Reply:
Removing NPE check.